synapse/config/registration.py

```diff
@@ -54,6 +55,8 @@ def default_config(self, **kwargs):
         # Larger numbers increase the work factor needed to generate the hash.
         # The default number of rounds is 12.
         bcrypt_rounds: 12
+
+        allow_guest_access: False
```

Comment:
It'd be nice to have a short explanation of what enabling guest access does here.